Source/CBL_Replicator.m

```diff
@@ -476,32 +480,73 @@ - (void) reachabilityChanged: (CBLReachability*)host {
     LogTo(Sync, @"%@: Reachability state = %@ (%02X), suspended=%d",
           self, host, host.reachabilityFlags, _suspended);
 
+    [self stopCheckRequest];
+
     BOOL reachable = host.reachable && !_suspended;
+    BOOL allowed = YES;
     if (reachable) {
-    // Parse "network" option. Could be nil or "WiFi" or "!Wifi" or "cell" or "!cell".
-    NSString* network = [$castIf(NSString, _options[kCBLReplicatorOption_Network])
+        // Parse "network" option. Could be nil or "WiFi" or "!Wifi" or "cell" or "!cell".
+        NSString* network = [$castIf(NSString, _options[kCBLReplicatorOption_Network])
                               lowercaseString];
         if (network) {
             BOOL wifi = host.reachableByWiFi;
             if ($equal(network, @"wifi") || $equal(network, @"!cell"))
-                reachable = wifi;
+                allowed = wifi;
             else if ($equal(network, @"cell") || $equal(network, @"!wifi"))
-                reachable = !wifi;
+                allowed = !wifi;
             else
                 Warn(@"Unrecognized replication option \"network\"=\"%@\"", network);
         }
     }
 
-    if (reachable)
+    if (reachable && allowed)
         [self goOnline];
-    else if (host.reachabilityKnown || _suspended)
+    else if (_suspended || !allowed)
         [self goOffline];
+    else if (host.reachabilityKnown)
+        [self mayGoOffline];
 }
 
+
+- (void) mayGoOffline {
```

Comment:
The name of this method is confusing to me ... Maybe make it -checkIfOffline? Or something like that?